### PR TITLE
chore(IDPay): [IODPAY-152] IDPay client refactoring with merged API definitions

### PR DESCRIPTION
**Pull request by mastro993**
## Short description

This PR introduces a single IDPay API client that combines all the IDPay API definitions. The old separated API clients was removed, resulting in a more streamlined and easy-to-understand code structure. 

## List of changes proposed in this pull request
- Removed old, splitted, IDPay APIs clients (wallet, initiative, onboarding, timeline)
- Added a new IDPay API client shared between all IDPay sub-features
- Refactoring of imports and folders to match new definitions
- Added `generate:idpay` script to generate IDPay API definitions

## How to test
`yarn test`

### CODE REVIEW

File: ts/features/design-system/core/DSListItems.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ import { ProductCategoryEnum } from "../../../../definitions/cgn/merchants/Produ
 import CgnMerchantListItem from "../../bonus/cgn/components/merchants/CgnMerchantListItem";
 import DetailedlistItemComponent from "../../../components/DetailedlistItemComponent";
 import { TimelineOperationListItem } from "../../idpay/initiative/details/components/TimelineOperationListItem";
-import { OperationTypeEnum } from "../../../../definitions/idpay/timeline/TransactionOperationDTO";
+import { OperationTypeEnum } from "../../../../definitions/idpay/TransactionOperationDTO";
```
Comment (francescopersico) [inline ts/features/design-system/core/DSListItems.tsx:25]: Just for reference. Is is not good to have the design system to depends on idpay or other specific features...

File: ts/features/idpay/initiative/configuration/xstate/provider.tsx [hunk 2]
```diff
@@ -94,14 +93,12 @@ const IDPayConfigurationMachineProvider = (props: Props) => {
 
   const pmSessionManager = new SessionManager(getPaymentManagerSession);
 
-  const walletClient = createIDPayWalletClient(
+  const idPayClient = createIDPayClient(
```
Comment (francescopersico) [inline ts/features/idpay/initiative/configuration/xstate/provider.tsx:96]: The problem here is that you are using a single client definition but still creating many clients for each flow.

File: ts/features/design-system/core/DSListItems.tsx (hunk 1 above)
Comment (mastro993) [reply ts/features/design-system/core/DSListItems.tsx:25]: cc @dmnplb

File: ts/features/idpay/initiative/configuration/xstate/provider.tsx (hunk 2 above)
Comment (mastro993) [reply ts/features/idpay/initiative/configuration/xstate/provider.tsx:96]: Yes that is true, we need to find an efficient way to provide a single client instance.

**Review by forrest57 — APPROVED** [review]

**Pull request merged**